src/doc/src/reference/build-scripts.md

```diff
@@ -403,6 +403,12 @@ variables like `TARGET` that [Cargo sets for build scripts][build-env]. The
 environment variables in use are those received by `cargo` invocations, not
 those received by the executable of the build script.
 
+As of 1.46, variables used by [`env!`][env-macro] and 
+[`option_env!`][option-env-macro] are automatically detected. So using a build
+script to register these with `rerun-if-env-changed` is not necessary.
```

Comment:
> My main concern when writing this was whether it was clear that this applies to env! macro usage within the main code of the crate and not just in build.rs.

What about this?

> As of 1.46, using `env!` and `option_env!` in source code will automatically detect changes and trigger rebuilds changes. `rerun-if-env-changed` is no longer needed for variables already referenced by these macros.

Reply:
Removing `changes` at the end of the first sentence since it looks like a typo:

> As of 1.46, using `env!` and `option_env!` in source code will automatically detect changes and trigger rebuilds. `rerun-if-env-changed` is no longer needed for variables already referenced by these macros.

The wording here looks like an improvement to me. Although I don't know if it alleviates my concern much, I would still be happy with this! I actually considered:
> So using a build script to register these with `rerun-if-env-changed`

to somewhat help imply that the macros used weren't specifically in a build script by focusing the fact that build scripts are involved on the need to use `rerun-if-env-changed`, but that may be a bit of a stretch.

Reply:
Understandable.

To me register is a more technical word that may need more explanation. Anyway, let me know if you want to do further revise. I could merge this as is as well.

Reply:
I'm going to change it to your suggested wording.